src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/PinnedFlowV2Spec.groovy

```diff
@@ -159,9 +160,18 @@ class PinnedFlowV2Spec extends HealthCheckSpecification {
             assert northboundV2.getFlowStatus(flow.flowId).status == FlowState.UP
             assert pathHelper.convert(northbound.getFlowPath(flow.flowId)) == currentPath
         }
+        def islsAreUp = true
 
-        and: "Cleanup: revert system to original state"
-        flowHelperV2.deleteFlow(flow.flowId)
+        cleanup:
+        flow && flowHelperV2.deleteFlow(flow.flowId)
+        if (islsToBreak && !islsAreUp) {
+            islsToBreak.each { islToRestore ->
+                antiflap.portUp(islToRestore.srcSwitch.dpId, islToRestore.srcPort)
+                Wrappers.wait(WAIT_OFFSET + discoveryInterval) {
+                    assert islUtils.getIslInfo(islToRestore).get().state == IslChangeType.DISCOVERED
+                }
```

Comment:
Not the most efficient way to restore in cleanup. In test these steps were intentionally used to restore isls one by one, but in cleanup it'll be better to request all portUps first and the do the waiting for isls.

Reply:
nice catch, fixed